### PR TITLE
[Workflow] Automatically Lint and Format PRs

### PR DESCRIPTION
**Pull request by benjaminye**
Addresses https://github.com/georgian-io/LLM-Finetuning-Hub/issues/103 and partially https://github.com/georgian-io/LLM-Finetuning-Hub/issues/106

## New Github Action
- Added Github action to Lint and Format using ruff
- Added configs for ruff
- Automatically commit code changes onto PR

## Lint and Format
- Linted and Formatted codebase (minus scripts within `llama2` and `mistral`)

## Request for Comment
@truskovskiyk

- Right now it's set up to automatically commit changes lint and formatting fixes directly onto the PR.
- Is this alright?
- This requires read/write access for github workflow runners under repo settings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -255,3 +255,10 @@ If you would like to contribute to this project, we recommend following the "for
 5.  Submit a **Pull request** so that we can review your changes
 
 NOTE: Be sure to merge the latest from "upstream" before making a pull request!
+
+### Checklist Before Pull Request (Optional)
+
+1. Use `ruff check` to check for lint errors
+2. Use `ruff format` to apply formatting
+
+NOTE: Ruff linting and formatting are done automatically when PR is raised using Git Action (and changes will be automatically applied via another commit). It is, however, a good practice to check and fix lint errors, as well as apply formatting before PR.
```
Comment (truskovskiyk) [inline README.md:264]: might be nice to add this into Makefile after

**Review by truskovskiyk — APPROVED** [review]
thanks, much nicer!

**Comment by truskovskiyk** [comment]
@benjaminye 

> Right now it's set up to automatically commit changes lint and formatting fixes directly onto the PR.
> Is this alright?


No, give access CI to write code into main branch is not best idea, let's just keep check-in CI - if code does not follow style - PR fails. but this exact fix - need to be done by author of PR

**Comment by benjaminye** [comment]
> @benjaminye
> 
> > Right now it's set up to automatically commit changes lint and formatting fixes directly onto the PR.
> > Is this alright?
> 
> No, give access CI to write code into main branch is not best idea, let's just keep check-in CI - if code does not follow style - PR fails. but this exact fix - need to be done by author of PR

@truskovskiyk 
Amended desired changes, as required. See https://github.com/georgian-io/LLM-Finetuning-Hub/pull/116/commits/36688e068b911708de64467c063410e5bb8e984d

**Pull request merged**